extension/datadogextension/internal/httpserver/httpserver.go

```diff
@@ -0,0 +1,206 @@
+// otel_collector payload as well as submit it to the Datadog backend.
+// Server provides local metadata server functionality (display the otel_collector payload locally) as well as
+// functions to serialize and export this metadata to Datadog backend.
+// Specifically, making a request to the local endpoint, configured in config.go, will both display the
+// otel_collector payload as well as submit it to the Datadog backend.
+type Server struct {
+	server               *http.Server
+	logger               *zap.Logger
+	serializer           serializer.MetricSerializer
+	forwarder            defaultForwarderInterface
+	cancel               context.CancelFunc
+	config               *Config
+	hostnameSource       string
+	hostname             string
+	uuid                 string
+	componentStatus      *map[string]any
+	moduleInfo           *service.ModuleInfos
+	collectorConfig      *confmap.Conf
+	otelCollectorPayload *payload.OtelCollector
+}
+
+// NewServer creates a new HTTP server instance
+func NewServer(
+	logger *zap.Logger,
+	config *Config,
+	s serializer.MetricSerializer,
+	f defaultForwarderInterface,
+	hs string,
+	hn string,
+	UUID string,
+	componentstatus *map[string]any,
+	moduleinfo *service.ModuleInfos,
+	collectorconfig *confmap.Conf,
+	collectorpayload *payload.OtelCollector,
+) *Server {
+	srv := &Server{
+		logger:               logger,
+		serializer:           s,
+		forwarder:            f,
+		config:               config,
+		hostnameSource:       hs,
+		hostname:             hn,
+		uuid:                 UUID,
+		componentStatus:      componentstatus,
+		moduleInfo:           moduleinfo,
+		collectorConfig:      collectorconfig,
+		otelCollectorPayload: collectorpayload,
+	}
+	return srv
+}
+
+// Start starts the HTTP server and begins sending payloads periodically
+func (s *Server) Start() {
+	ctx, cancel := context.WithCancel(context.Background())
+	s.cancel = cancel
+
+	// Create a mux to handle only the specific path
+	mux := http.NewServeMux()
+	mux.HandleFunc(s.config.Path, s.HandleMetadata)
+
+	s.server = &http.Server{
+		Addr:         s.config.Endpoint,
+		Handler:      mux,
+		ReadTimeout:  5 * time.Second,
+		WriteTimeout: 10 * time.Second,
+		BaseContext:  func(net.Listener) context.Context { return ctx },
+	}
+
+	// Start HTTP server
+	go func() {
+		if err := s.server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			s.logger.Error("HTTP server error", zap.Error(err))
+		}
+	}()
+
+	// Monitor context cancellation
+	go func() {
+		<-ctx.Done()
+		if err := s.server.Shutdown(context.Background()); err != nil {
+			s.logger.Error("Error during server shutdown", zap.Error(err))
+		}
+	}()
+
+	s.logger.Info("HTTP Server started at " + s.config.Endpoint + s.config.Path)
+}
+
+// Stop shuts down the HTTP server
+func (s *Server) Stop() {
+	if s.server != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
```

Comment:
nit: typically I would handle this in the caller, by pasing a `context.Context` struct to the `Stop` method